src/scrcmd_c.c

```diff
@@ -843,3 +862,1020 @@ BOOL ScrCmd_061(SCRIPTCONTEXT* ctx) {
+    GF_ASSERT(r4 != NULL);
+    int r0 = sub_0203E5C8(*p_work);
+    if (r0 == 7) {
+        *r5 = 255;
+    } else if (r0 == 6) {
+        *r5 = r4->unk_30;
+        (*r5)--;
+        *sp0 = r4->unk_31;
+        (*sp0)--;
+        *r7 = r4->unk_32;
+        if (*r7 != 0) {
+            (*r7)--;
+        }
+    }
+    FreeToHeap(*p_work);
+    *p_work = NULL;
+    return FALSE;
+}
+
+BOOL ScrCmd_GetMoveSelection(SCRIPTCONTEXT *ctx) {
+    u8 r6 = ScriptReadByte(ctx);
+    u16 *r5 = ScriptGetVarPointer(ctx);
+    void **p_work = FieldSysGetAttrAddr(ctx->fsys, SCRIPTENV_AC);
+    GF_ASSERT(*p_work != NULL);
+    if (r6 == 1) {
+        *r5 = sub_0203E864(*p_work);
+    } else {
+        *r5 = sub_0203E600(*p_work);
+    }
+    FreeToHeap(*p_work);
+    *p_work = NULL;
+    return FALSE;
+}
+
+BOOL ScrCmd_ChooseMoveUI(SCRIPTCONTEXT *ctx) {
+    void **p_work = FieldSysGetAttrAddr(ctx->fsys, SCRIPTENV_AC);
+    u8 r6 = ScriptReadByte(ctx);
+    u16 r7 = ScriptGetVar(ctx);
+    u16 r3 = ScriptGetVar(ctx);
+    if (r6 == 1) {
+        *p_work = sub_0203E7F4(32, ctx->fsys, r7, r3);
+    } else {
+        *p_work = sub_0203FB94(32, ctx->fsys, r7, r3);
+    }
+    SetupNativeScript(ctx, ScrNative_WaitApplication);
+    return TRUE;
+}
+
+BOOL ScrCmd_GetPhoneBookRematch(SCRIPTCONTEXT *ctx) {
+    u16 r4 = ScriptGetVar(ctx);
+    u16 *r6 = ScriptGetVarPointer(ctx);
+    struct PhoneBook *phoneBook = AllocAndReadPhoneBook(32);
+    HandleLoadOverlay(FS_OVERLAY_ID(OVY_26), OVY_LOAD_ASYNC);
+    *r6 = PhoneBookTrainerGetRematchInfo(r4, ctx->fsys->savedata, phoneBook, Field_GetTimeOfDayWildParam(ctx->fsys));
+    UnloadOverlayByID(FS_OVERLAY_ID(OVY_26));
+    FreePhoneBook(phoneBook);
+    return FALSE;
+}
+
+BOOL ScrCmd_684(SCRIPTCONTEXT *ctx) {
+    u16 *p_dest = ScriptGetVarPointer(ctx);
+    *p_dest = FlyPoints_GetWeatherType(Save_FlyPoints_get(ctx->fsys->savedata));
+    return FALSE;
+}
+
+BOOL ScrNative_WaitApplication_DestroyTaskData(SCRIPTCONTEXT *ctx) {
+    FieldSystem *fsys = ctx->fsys;
+    void **p_work = FieldSysGetAttrAddr(fsys, SCRIPTENV_AC);
+    if (FieldSys_ApplicationIsRunning(fsys)) {
+        return FALSE;
+    }
+    FreeToHeap(*p_work);
+    *p_work = NULL;
+    return TRUE;
+}
+
+BOOL sub_020429A0(SCRIPTCONTEXT *ctx) {
+    FieldSystem *fsys = ctx->fsys;
+    void **p_work = FieldSysGetAttrAddr(fsys, SCRIPTENV_AC);
+    if (FieldSys_ApplicationIsRunning(fsys)) {
+        return FALSE;
+    }
+    sub_02093070(fsys);
+    FreeToHeap(*p_work);
+    *p_work = NULL;
+    return TRUE;
+}
+
+BOOL sub_020429D4(SCRIPTCONTEXT *ctx) {
+    FieldSystem *fsys = ctx->fsys;
+    void **p_work = FieldSysGetAttrAddr(fsys, SCRIPTENV_AC);
+    if (!sub_020970C0(*p_work)) {
+        return FALSE;
+    }
+    *p_work = NULL;
+    return TRUE;
+}
+
+BOOL ScrNative_WaitApplication(SCRIPTCONTEXT *ctx) {
+    if (!FieldSys_ApplicationIsRunning(ctx->fsys)) {
+        return TRUE;
+    } else {
+        return FALSE;
+    }
```

Comment:
```suggestion
    return !FieldSys_ApplicationIsRunning(ctx->fsys);
```
`return FieldSys_ApplicationIsRunning(ctx->fsys) == FALSE;` also works

Reply:
doesn't match tho